contentcuration/contentcuration/frontend/channelEdit/views/progress/ProgressModal.vue

```diff
@@ -108,19 +109,52 @@
         type: String,
         default: '',
       },
+      syncing: {
+        type: Boolean,
+        default: false,
+      },
+      noSyncNeeded: {
+        type: Boolean,
+        default: false,
+      },
     },
     data() {
       return {
         step: 1,
       };
     },
     computed: {
-      ...mapGetters('task', ['publishTaskForChannel']),
+      ...mapGetters('task', ['currentTasksForChannel']),
       ...mapGetters('currentChannel', ['currentChannel']),
+      currentTasks() {
+        return this.currentTasksForChannel(this.currentChannel.id) || null;
+      },
+      isSyncing() {
+        return this.syncing && !this.currentChannel.publishing;
```

Comment:
Do you need to check that `this.currentChannel` before accessing `.publishing`, like in `isPublishing`?

Reply:
That is a good question. The check for publishing was originally in a `v-if` on the modal, and so I kept that while I was refactoring. I used both (as you suggest) when I was working on it locally, and then thought maybe it was redundant. My reasoning was that the falsey value from `this.currentChannel.publishing` would be sufficient, but maybe it would be better to include it...

Reply:
Yeah, I would just copy the implementation of `isPublishing` to future-proof the code and avoid type errors. It would also make it clearer that the type of `this.currentChannel` is null or an object.

Reply:
```suggestion
        return this.syncing && this.currentChannel && !this.currentChannel.publishing;
```

Reply:
Yes, good idea. Thanks 🙏  updated!